docs/docs/domain.mdx

```diff
@@ -465,64 +470,180 @@ stories:
+The `from_intent` mapping will fill slot `slot_name` with value `my_value` if
+user intent is `intent_name` or `None`. The slot mapping will not
+apply if the intent of the message is `excluded_intent`. Note that you can
+also define lists of intents for the parameters `intent` and `not_intent`.
 
-In the example above, if the `name` slot's type is `text`, then it only matters that some name was detected, but it won't matter which one.
-If the `name` slot is type `categorical`, then the behaviour will vary depending on the categories you have defined for the slot. 
+```yaml-rasa
+slots:
+  slot_name:
+    type: any
+    mappings:
+    - type: from_intent
+      value: my_value
+      intent: intent_name
+      not_intent: excluded_intent
+```
 
-Explicitly including a `slot_was_set` step in your story can make the behaviour of an
-auto-filled slot that influences the conversation clearer, and will not change the behaviour of your story.
+#### from_trigger_intent
 
-:::
+The `from_trigger_intent` mapping will fill slot `slot_name` with value `my_value`
+if a form is activated by a user message with intent `intent_name`.
+The slot mapping will not apply if the intent of the message is
+`excluded_intent`. Note that you can
+also define lists of intents for the parameters `intent` and `not_intent`.
+
+A slot with `from_trigger_intent` mapping must always include mapping conditions
+and be listed in the form's `required_slots`.
+
+```yaml-rasa
+slots:
+  slot_name:
+    type: any
+    mappings:
+    - type: from_trigger_intent
+      value: my_value
+      intent: intent_name
+      not_intent: excluded_intent
+      conditions:
+      - active_loop: your_form
+forms:
+  your_form:
+    required_slots:
+    - slot_name
+```
 
+### Mapping Conditions
 
+If you prefer to extract and set a slot only within the context of a form, you can indicate
+the name of the form in the `conditions` key of a mapping. Conditions list the form name(s)
+for which the mapping is applicable in the `active_loop` key.
 
-To disable auto-filling for a particular slot, you can set the
-`auto_fill` attribute to `false` in the domain file:
+Conditions can also include the name of the `requested_slot`. If `requested_slot` is not mentioned,
+then the slot will be set if relevant information is extracted, regardless of which slot is being
+requested by the form.
 
 ```yaml-rasa
 slots:
-  name:
+  slot_name:
     type: text
     mappings:
-    - type: from_entity
-      entity: name
+    - type: from_text
+      intent: intent_name
+      conditions:
+      - active_loop: your_form
+        requested_slot: slot_name
+      - active_loop: another_form
+```
+
+:::note
+If conditions are not included in a slot mapping, the slot is eligible to be set universally
+outside the context of any forms. However if a slot is also listed in a form's `required_slots`,
+the slot can also be requested by the form if unset at that moment.
```

Comment:
I wanted to emphasise the behaviour without `conditions` so that there are no surprises if this key is omitted. And also to point out that forms cannot update slots, they can only set slots if they are unset at that point in the conversation.